src/pybind/mgr/cephadm/module.py

```diff
@@ -2425,6 +2429,15 @@ def _invalidate_daemons_and_kick_serve(self, filter_host: Optional[str] = None)
 
         self._kick_serve_loop()
 
+    def _invalidate_all_host_metadata_and_kick_serve(self, filter_host: Optional[str] = None) -> None:
+        if filter_host:
+            self.cache.refresh_all_host_info(filter_host)
+        else:
+            for h in self.cache.get_hosts():
+                self.cache.refresh_all_host_info(h)
```

Comment:
Just for my understanding: why are we refreshing all the hosts? aren't we supposed to refresh only the host which is about to go into maintenance mode?

Reply:
yeah, it should only run this for loop if `filter_host` is `None` and in the call in the maintenance exit command we pass in the hostname of the host being taken out of maintenance mode.

Reply:
Do you really need the `filter_host: Optional[str] = None` if this is *always* going to be called with a filter_host str?  IOW, make filter_host mandatory and eliminate the branch that will never get called in practice. If you need to add a version that passes `filter_host = None` in the future you can (re)add the if-else then.